index.tsx

```diff
@@ -89,32 +100,66 @@ export class StaticRouters {
         headers: { Location: result.redirect },
       });
     }
-    const stream = await renderToReadableStream(
-      <Shell route={serverSide.pathname + search} {...result}>
-        <module.default {...result?.props} />
-      </Shell>,
-      {
-        signal: request.signal,
-        bootstrapScriptContent: [
-          preloadScript,
-          `__PAGES_DIR__=${JSON.stringify(this.pageDir)}`,
-          `__INITIAL_ROUTE__=${JSON.stringify(serverSide.pathname + search)}`,
-          `__ROUTES__=${this.#routes_dump}`,
-          `__SERVERSIDE_PROPS__=${stringified}`,
-        ]
-          .filter(Boolean)
-          .join(";"),
-        bootstrapModules,
-        onError,
-      }
-    );
+    let jsxToServe: JSX.Element = <></>;
+    switch (this.options.displayMode) {
+      case "none":
+        jsxToServe = (
+          <Shell route={serverSide.pathname + search} {...result}>
+            <module.default {...result?.props} />
+          </Shell>
+        );
+      case "nextjs":
+        jsxToServe = await this.stackLayouts(serverSide);
+        break;
+    }
+
+    const stream = await renderToReadableStream(jsxToServe, {
+      signal: request.signal,
+      bootstrapScriptContent: [
+        preloadScript,
+        `__PAGES_DIR__=${JSON.stringify(this.pageDir)}`,
+        `__INITIAL_ROUTE__=${JSON.stringify(serverSide.pathname + search)}`,
+        `__ROUTES__=${this.#routes_dump}`,
+        `__SERVERSIDE_PROPS__=${stringified}`,
+      ]
+        .filter(Boolean)
+        .join(";"),
+      bootstrapModules,
+      onError,
+    });
     return new Response(stream, {
       headers: {
         "Content-Type": "text/html; charset=utf-8",
         "Cache-Control": "no-store",
       },
     });
   }
+  /**
+   * Next.js like module stacking
+   */
+  async stackLayouts(route: MatchedRoute) {
+    const layouts = route.pathname.split("/").slice(1);
+    let layoutsJsxList: React.ComponentType<{
+      children: React.ReactElement;
+    }>[] = [];
+    let index = 0;
+    for await (const i of layouts) {
+      const path = layouts.slice(0, index).join("/");
+      const pathToFile = `${this.baseDir}/${this.pageDir}${path}${this.options.layoutName}`;
+      console.log(path, index);
+      if (!(await Bun.file(pathToFile).exists())) continue;
+      const defaultExport = (await require(pathToFile)).default;
```

Comment:
shouldn't be import? (yes I know require works in esm mode, but it is not async, and may not working with top level await)

Reply:
i cannot say why but import could not be resolve for some reason

Reply:
now import